cloud/amazon/ec2_ami.py

```diff
@@ -225,6 +225,12 @@ def create_image(module, ec2):
                   'description': description,
                   'no_reboot': no_reboot}
 
+        images = ec2.get_all_images(filters={'name': name})
+
+        if images and images[0]:
+            module.exit_json(msg="AMI name already present", image_id=images[0].id, state=images[0].state, changed=False)
+            sys.exit(0)
```

Comment:
my other thought is that, this might just need to return success and changed=false, but that would require making sure that the existing AMI is the same as one would want to create